src/include/storage/wal/wal_record.h

```diff
@@ -68,6 +68,11 @@ struct WALRecord {
 
     virtual void serialize(common::Serializer& serializer) const;
     static std::unique_ptr<WALRecord> deserialize(common::Deserializer& deserializer);
+
+    template<class TARGET>
+    const TARGET& constCast() const {
```

Comment:
I think you can just make it cast() without const.

And for caller you could do
```
const auto& a = base.cast<>()
```

to obtain const reference